tests/conftest.py

```diff
@@ -1,12 +1,33 @@
 import os
 
 import pytest
+from dbt.tests.util import write_file
+from yaml import SafeDumper, dump_all
 
 # Import the standard functional fixtures as a plugin
 # Note: fixtures with session scope need to be local
 pytest_plugins = ['dbt.tests.fixtures.project']
 
 
+class Secret:
```

Comment:
Since we require pydantic in python sdk, can we use it's `SecretStr` implementation?
https://docs.pydantic.dev/latest/examples/secrets/

Reply:
Sure, refactored to use it. We still require a custom dumper but now there's no need to have our own Secret class.